### DISCUSSION TITLE
NextJS Deploy di AWS

### DISCUSSION THREAD

**Discussion by capiones** [awsugid/forum#27] [General]
Hallo, saya berniat buat deploy web saya di AWS. untuk tech yang saya gunakan itu adalah
- NextJS 14 menggunakan server side juga
- PostgreSQL untuk database nya
- ExpressJS/Golang sebagai Backend nya (ini belum jadi)
- Rencananya bakal gunakan Docker

apa yang harus saya lakukan biar bisa publish web saya tersebut? dan kira kira berapa biaya yang diperlukan?

**Comment by rioastamal** [comment]
@capiones,

- NextJS bisa menggunakan AWS Amplify
- PostgreSQL bisa menggunakan layanan managed seperti Amazon RDS atau install sendiri PostgreSQL di EC2. Tergantung preferensi.
- ExpressJS/Golang bisa menggunakan Amazon ECS/Amazon EKS/AWS Lambda

Atau bisa baca-baca di: https://aws.amazon.com/blogs/mobile/amplify-next-js-13/
Keywords nextjs: https://aws.amazon.com/blogs/mobile/tag/next-js/

**Reply by danielcristho** [reply]
Wah, dapat juga nih referensi buat deploy NextJS pakai Amplify. Thanks mas Rio.

**Comment by iilness2** [comment]
bisa juga deploy nextjsnya ke s3. 

Untuk langkah2nya pernah tak post disini: https://dev.to/iilness2/practical-approach-deploy-nextjs-with-s3-4k3l
